### PR TITLE
Implementation of Keyword Extraction in Javascript

### PR DESCRIPTION
**Pull request by jornichol**

### CODE REVIEW

File: project/src/main/webapp/keyword.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        return 'None';
+    }
+}
+
+async function keywordExtraction(userInput) {
+    var result = "";
+    var count = 0;
+    var searchQuery = userInput.split(" ");
+    for (word of searchQuery){
+        if ((searchQuery.length-1) == count && word.indexOf('?') != -1) {
```
Comment (eswarty) [inline project/src/main/webapp/keyword.js]: Nit: Add spacing in `searachQuery.length - 1`

File: project/src/main/webapp/keyword.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+function editedKeywordExtraction(userInput) {
```
Comment (eswarty) [inline project/src/main/webapp/keyword.js]: Please add comments for each function and give them more descriptive names (e.g. datamuseKeyWordExtraction vs naiveKeywordExtraction or maybe getQuestionType is more appropriate?)

File: project/src/main/webapp/script.js [hunk 3]
```diff
@@ -1,28 +1,21 @@
...
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+async function keywordExtraction(userInput) {
```
Comment (eswarty) [inline project/src/main/webapp/script.js]: Remove duplicate function

File: project/src/main/webapp/keyword.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    var searchQuery = userInput.split(" ");
+    for (word of searchQuery){
+        if ((searchQuery.length-1) == count && word.indexOf('?') != -1) {
+                word = word.substring(0, word.length-1);
+            }
+        var url = new URL("https://api.datamuse.com/words?md=p&sp=" + word + "&max=1");
+        const output = await fetch(url).then(response => response.text());
+        if (output.includes("[\"n\"]") == true || output.includes("[\"adj\"]") == true){
+            result += word;
+            if (count != (searchQuery.length-1)){
```
Comment (eswarty) [inline project/src/main/webapp/keyword.js]: Nit: Fix spacing here too

**Review by eswarty — CHANGES_REQUESTED** [review]

**Review by brandonvu12 — APPROVED** [review]

**Review by eswarty — APPROVED** [review]

**Pull request merged**